src/settings.py

```diff
@@ -206,8 +206,10 @@ def get_arg_specific_settings(args: Dict[str, Any]) -> Dict[str, Any]:
 def get_allowed_query_type_names() -> Set[str]:
     """Get the set of names of query types that the teacher is allowed to
     answer, computed based on the configuration CFG."""
-    if CFG.approach in ("interactive_learning", "unittest"):
```

Comment:
not quite... the logic should be:
* if neural option learner, then {"DemonstrationQuery"}
* if interactive learning approach, then {"GroundAtomsHoldQuery"}
* if unittest, then both (this is fine as is)